lnwallet/wallet.go

```diff
@@ -446,11 +452,25 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 		return
 	}
 
+	// If no chanFunder was provided, then we'll assume the default
+	// assembler, which is backed by the wallet's internal coin selection.
+	if req.ChanFunder == nil {
```

Comment:
Isn't a better separation to let the  creator of `InitFundingReserveMsg` set this? That way we can (slowly) move towards separating this channel funding code from the wallet.

Reply:
As is it does, but the idea here it to provide a default so the caller doesn't have to. In this case, the caller is the funding manager, which isn't _yet_ aware of the new channel funding abstractions. Having the default here also meshes well into the existing funding workflow tests in `interface_test.go` as well. 

In the end, we call this file `wallet.go`, but it's almost purely implementing the funding workflow.